Fix squeeze inside NeuralNetBinaryClassifier.infer

qtux opened the pull request:
Previously y_infer.squeeze() removed all dimensions with single entries,
which does not correspond to the docstring description. Instead, remove
only the last dimension, if it contains a single entry.

BenjaminBossan commented:
Thanks for the PR. I think this change should be okay.

Could you please add an entry to the CHANGES.md?

qtux commented:
Sure, I will rebase the commit.

qtux commented:
I added a test to show the bug and replaced the squeeze by a reshape (was not correct previously).

ottonemo commented on skorch/tests/test_classifier.py:
```diff
@@ -235,6 +235,16 @@ def test_net_learns(self, net_cls, module_cls, data):
 
+    def test_batch_size_one(self, net_cls, module_cls, data):
+        X, y = data
+        print("test", y.shape)
```
```suggestion
```

qtux commented:
I removed the print statement from the test and rebased the commit.

On 15/11/2019 17:13, ottonemo wrote:
> *@ottonemo* approved this pull request.
> 
> ------------------------------------------------------------------------
> 
> In skorch/tests/test_classifier.py 
> <https://github.com/skorch-dev/skorch/pull/558#discussion_r346894482>:
> 
>> @@ -235,6 +235,16 @@ def test_net_learns(self, net_cls, module_cls, data):
>           valid_acc = net.history[-1, 'valid_acc']
>           assert valid_acc > 0.65
>   
> +    def test_batch_size_one(self, net_cls, module_cls, data):
> +        X, y = data
> +        print("test", y.shape)
> 
> ⬇️ Suggested change
> 
> -        print("test", y.shape)
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/skorch-dev/skorch/pull/558?email_source=notifications&email_token=ABKQM3YONALGCA62AOI4CPTQT3DEPA5CNFSM4JMFX6EKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCLX64EQ#pullrequestreview-317713938>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ABKQM32MW3NLRPM2PX7KYPDQT3DEPANCNFSM4JMFX6EA>.
>

BenjaminBossan commented:
Thank you very much for the bugfix.
